### PR TITLE
chore: consolidate TestResult logic

### PR DESCRIPTION
**Pull request by grandizzy**
## Motivation
Creation of `TestResult` is handled in multiple places, PR is meant to consolidate that in `TestResult` struct and remove duped code from test runner

### CODE REVIEW

File: crates/forge/src/result.rs [hunk 1]
```diff
@@ -416,10 +421,170 @@ impl fmt::Display for TestResult {
...
+            logs: setup.logs,
+            traces: setup.traces,
+            coverage: setup.coverage,
+            labeled_addresses: setup.labeled_addresses,
+            ..Default::default()
+        }
+    }
+
+    /// Returns the skipped result for single test (used in skipped fuzz test too).
+    pub fn single_skip(&mut self) -> Self {
```
Comment (DaniPopes) [inline crates/forge/src/result.rs:455]: these should all take self by value `mut self`

File: crates/forge/src/result.rs [hunk 2]
```diff
@@ -429,6 +594,17 @@ impl TestResult {
...
     }
+
+    /// Function to merge given coverage in current test result coverage.
+    fn merge_coverages(&mut self, other_coverage: Option<HitMaps>) {
+        let old_coverage = std::mem::take(&mut self.coverage);
+        self.coverage = match (old_coverage, other_coverage) {
+            (Some(old_coverage), Some(other)) => Some(old_coverage.merged(other)),
+            (None, Some(other)) => Some(other),
+            (Some(old_coverage), None) => Some(old_coverage),
+            (None, None) => None,
```
Comment (DaniPopes) [inline crates/forge/src/result.rs:605]:
```suggestion
            (a, b) => a.or(b),
```

**Review by DaniPopes — APPROVED** [review]

File: crates/forge/src/result.rs (hunk 1 above)
Comment (grandizzy) [reply crates/forge/src/result.rs]: changed in https://github.com/foundry-rs/foundry/pull/8208/commits/98e0f1bac11a83459467270fb884bf60fa752941

File: crates/forge/src/result.rs [hunk 3]
```diff
@@ -416,10 +421,170 @@ impl fmt::Display for TestResult {
...
+            labeled_addresses: setup.labeled_addresses,
+            ..Default::default()
+        }
+    }
+
+    /// Returns the skipped result for single test (used in skipped fuzz test too).
+    pub fn single_skip(mut self) -> Self {
+        self.status = TestStatus::Skipped;
+        self.decoded_logs = decode_console_logs(&self.logs);
+        self.to_owned()
```
Comment (DaniPopes) [inline crates/forge/src/result.rs:458]: and with self by value the `to_owned` call is superfluous, pls remove

Comment (grandizzy) [reply crates/forge/src/result.rs:458]: Oppsie, will change

**Pull request merged**